src/ips/messages.ts

```diff
@@ -33,12 +34,12 @@ export class Messages {
     return this.wsmanMessageCreator.createXml(header, body)
   }
 
-  private readonly put = (action: AllActions, ipsClass: Classes, data: OptInServiceResponse): string => {
+  private readonly put = (action: AllActions, ipsClass: Classes, data: any): string => {
```

Comment:
why `any` when it was typed before?

Reply:
both ieee8021xSettings and OptInServiceResponse object types need to flow through this.  I've added the type check at the specific function level for those calls.  The private put call will to be more generic and accept different data types.